### PR TITLE
Convert to native nomad sd

### PR DESCRIPTION
**Pull request by attachmentgenie**
based on PR #40 , potentially blocked by hashicorp/nomad#13483

decided to see what it would take to converted this example to only use the native SD, had to convert to traefik, but since that is already in use for the cloud examples i thought that was ok

### CODE REVIEW

File: vagrant/horizontal-app-scaling/jobs/traefik.nomad
Comment (lgfa29) [inline vagrant/horizontal-app-scaling/jobs/traefik.nomad:42]: I think it would be good to keep `network_mode = "host` here so that we can access Traefik using `127.0.0.1` as currently [defined in the instructions](https://learn.hashicorp.com/tutorials/nomad/autoscaler-vagrant-demo?in=nomad/autoscaler#generate-load-and-scale-up).

**Review by lgfa29 — CHANGES_REQUESTED** [review]
Thanks for this one as well @attachmentgenie!

I gave a try and it's looking good. I just made a comment about the `network_mode` so it's easier to write generic instructions.

Another thing is that it may be better to hold off a bit on this change until health check is available for Nomad native service discovery. Then I think this will look great 😄

File: vagrant/horizontal-app-scaling/jobs/traefik.nomad
Comment (attachmentgenie) [reply vagrant/horizontal-app-scaling/jobs/traefik.nomad:42]: sure i am willing to wait for the checks to be part of nomad native sd. I ll see how far i get with nomad 1.3.5. Will fix this one, don’t think i did this deliberately, prob just a copy paste error.

File: vagrant/horizontal-app-scaling/jobs/webapp.nomad
Comment (lgfa29) [inline vagrant/horizontal-app-scaling/jobs/webapp.nomad:22]:
```suggestion
          query  = "sum(traefik_entrypoint_open_connections{entrypoint=\"webapp\"} OR on() vector(0))/scalar(nomad_nomad_job_summary_running{task_group=\"demo\"})"
```

I think we need this to cause the Autoscaler to bring count to 1 before running `hey`. From the tutorial:

> The autoscaler observes that the application is idle and scales it down to the min count of 1.

Without this Prometheus returns no data instead of zero, and the Autoscaler errors with `no metrics available`. Could you check if this fixes the problem for you as well?

**Review by lgfa29 — CHANGES_REQUESTED** [review]

File: vagrant/horizontal-app-scaling/jobs/webapp.nomad
Comment (attachmentgenie) [reply vagrant/horizontal-app-scaling/jobs/webapp.nomad]: nice find, accepted!

**Review by lgfa29 — APPROVED** [review]
Thanks for the work @attachmentgenie!

I pushed a small commit to add the job name in the query. While updating the tutorial text I noticed that not having it could be confusing when explaining how the query works.

I have a PR open against the internal repo where the instructions live. Once that's merged I will update the `learn` tag to pick this up 🙂

**Pull request merged**
